src/tools/crossgen2/Common/TypeSystem/Interop/IL/MarshalUtils.cs

```diff
@@ -13,47 +13,48 @@ public static class MarshalUtils
         /// </summary>
         public static bool IsBlittableType(TypeDesc type)
         {
-            type = type.UnderlyingType;
+            if (!type.IsDefType)
+            {
+                return false;
+            }
+
+            TypeDesc baseType = type.BaseType;
+            bool hasNonTrivialParent = baseType != null
```

Comment:
Is this dance necessary? I would expect that blittable is false for `!type.IsValueType`

Reply:
Reference types can also be blittable. E.g. WinForms is full of those (all the p/invokes that a sane person would do as `ref SomeStruct` or `SomeStruct*` parameter, WinForms instead makes them just `SomeClass`). We don't have those marshallers in the R2R compiler, but they exist and they rely on this API to work like this.

Existing places that call `IsBlittable` on the R2R side [filter down to value types](https://github.com/dotnet/coreclr/blob/fa4a86b1dc5aa021981df70ffd628d68e100beb5/src/tools/crossgen2/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs#L1729) first, but the API should return true for blittable classes.

Reply:
(I'll have to fix up the callsites to IsBlittable on the CoreRT side when I port this over because some of those callsites rely on the wrong definition of IsBlittable that implies the type is a struct.)

Reply:
The comment on the API says "Returns true if this is a type that doesn't require marshalling.". It is not quite right then.

Reply:
https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types says "Object references are not blittable.". It may be a good idea to stick to standard definition of blittable.

Reply:
Scratch the last comment. I see that you are aligning to the standard definition.

Reply:
Updated the comment to more accurately describe what this is.